src/scripts/main.js

```diff
@@ -1,3 +1,291 @@
+    position: 'Regional Director',
+    office: 'Edinburgh',
+    age: 51,
+    salary: '$183,000',
+  },
+  {
+    name: 'Shou Itou',
+    position: 'Regional Marketing',
+    office: 'Tokyo',
+    age: 20,
+    salary: '$163,000',
+  },
+  {
+    name: 'Vivian Harrell',
+    position: 'Financial Controller',
+    office: 'San Francisco',
+    age: 62,
+    salary: '$452,500',
+  },
+  {
+    name: 'Zorita Serrano',
+    position: 'Software Engineer',
+    office: 'San Francisco',
+    age: 56,
+    salary: '$115,000',
+  },
+  {
+    name: 'Colleen Hurst',
+    position: 'Javascript Developer',
+    office: 'San Francisco',
+    age: 39,
+    salary: '$205,500',
+  },
+];
+
+function initTableBody() {
+  tableBody.innerHTML = `
+    <tbody>
+      ${usersData.map(item => `
+        <tr>
+          <td>${item.name}</td>
+          <td>${item.position}</td>
+          <td>${item.office}</td>
+          <td>${item.age}</td>
+          <td>${item.salary}</td>
+        </tr>
+      `).join('')}
+    </tbody>
+  `;
+}
+
+function initForm() {
+  table.insertAdjacentHTML('afterend', `
+    <form class="new-employee-form">
+      <label>Name:
+        <input name="name" type="text" data-qa="name">
+      </label>
+      <label>Position:
+        <input name="position" type="text" data-qa="position">
+      </label>
+      <label>Office:
+        <select name="office" data-qa="office">
+          <option value="San Francisco">San Francisco</option>
+          <option value="Tokyo">Tokyo</option>
+          <option value="Singapore">Singapore</option>
+          <option value="London">London</option>
+          <option value="New York">New York</option>
+          <option value="Edinburgh">Edinburgh</option>
+        </select>
+      </label>
+      <label>Age:
+        <input name="age" type="number" data-qa="age">
+      </label>
+      <label>Salary:
+        <input name="salary" type="number" data-qa="salary">
+      </label>
+      <button type="submit">Save to table</button>
+    </form>
+  `);
+}
+
+function initNotification(type, message) {
+  notification.classList.add('notification', type);
+
+  notification.innerHTML = `
+    <p class="title">
+      ${message}
+    </p>
+  `;
+}
+
+function validate(key, value) {
+  if (!value) {
+    return false;
+  }
+
+  if (key === 'name' && value.length < 4) {
+    return false;
+  }
+
+  if (key === 'age' && (value < 18 || value > 90)) {
+    return false;
+  }
+
+  return true;
+}
+
+const table = document.querySelector('table');
+const tableHead = table.querySelector('thead');
+const tableBody = document.querySelector('tbody');
+
+initTableBody();
+initForm();
+
+const form = document.querySelector('form');
+const notification = document.querySelector('[data-qa="notification"]');
+
+let index;
+let count = 1;
+
+tableHead.addEventListener('click', (e) => {
+  const item = e.target.textContent.toLowerCase();
+
+  const dir = count % 2 === 0 && item === index
+    ? 'desc'
+    : 'abs';
```

Comment:
What does it mean `abs`?
```suggestion
    : 'asc';
```